src/thunderbird_accounts/subscription/views.py

```diff
@@ -23,6 +35,96 @@ def prefilter_paddle_webhook(event_type: str, event_data: dict) -> bool:
+    for plan in plans:
+        prices = plan.product.price_set.filter(status=Price.StatusValues.ACTIVE).all()
+        plan_info.append({'name': plan.name, 'prices': [price.paddle_id for price in prices]})
+        plan_info.extend([price.paddle_id for price in prices])
+
+    return JsonResponse(
+        {
+            'paddle_token': settings.PADDLE_TOKEN,
+            'paddle_environment': settings.PADDLE_ENV,
+            'paddle_plan_info': plan_info,
+            'signed_user_id': signer.sign(request.user.uuid.hex),
+        }
+    )
+
+
+@login_required
+@require_http_methods(['POST'])
+@inject_paddle
+def notify_paddle_checkout_complete(request: Request, paddle: Client):
+    data = json.loads(request.body.decode())
+    transaction_id = data.get('transactionId')
+
+    print(data, transaction_id, request.body)
+    if not transaction_id:
+        return JsonResponse({'success': False})
+
+    # Give paddle some time to update their end
+    # Yes this is not perfect, I know.
+    time.sleep(5)
+
+    # We can't get webhook events on dev, so we need to manually check the event stream
+    if settings.IS_DEV:
+        subscription_id = None
+        notification_list = paddle.notifications.list(ListNotifications(filter=transaction_id))
+
+        # First look for the transaction
+        for notification in notification_list:
+            # Normalize the data to dict then a json blob then back to a dict.
+            notification_data = json.loads(paddle.serialize_json_payload(notification.payload.data.to_dict()))
+            if (
+                notification_data.get('id') == transaction_id
+                and notification.type == 'transaction.updated'
+                and notification_data.get('status') == 'completed'
+            ):
+                tasks.paddle_transaction_event.run(notification_data, notification.occurred_at, is_create_event=True)
+                subscription_id = notification_data.get('subscription_id')
+                break
+
+        # If we've found the transaction then we should have the subscription_id
+        if subscription_id:
+            subscription_list = paddle.notifications.list(ListNotifications(filter=subscription_id))
+
+            # Look up the subscription now
+            for notification in subscription_list:
+                notification_data = json.loads(paddle.serialize_json_payload(notification.payload.data.to_dict()))
+                # The serialization trick is not perfect, we need to flatten custom_data
+                notification_data['custom_data'] = notification_data.get('custom_data', {}).get('data', {})
+
+                if notification_data.get('id') == subscription_id and notification.type == 'subscription.created':
+                    tasks.paddle_subscription_event.run(
+                        notification_data,
+                        notification.occurred_at,
+                        is_create_event=True,
+                    )
+                    break
```

Comment:
Notifications are just the webhook record / payload accessible via api. Very handy, because this would otherwise be super annoying to do.